gnomad/resources/grch37/reference_data.py

```diff
@@ -270,3 +270,62 @@ def get_truth_ht() -> hl.Table:
         .repartition(200, shuffle=False)
         .persist()
     )
+
+
+def get_clinvar_pathogenic(version: str = None) -> hl.Table:
```

Comment:
I think we might want to rethink this a little. Currently the same function is basically copied for 38 and 37, but the actual filtering really doesn't need to be in each resource file. I wonder if we should add a function to utils/filtering and call that instead. I am interested to hear what @mike-w-wilson  thinks too since he handled all the reorganization. So I am thinking add something like this to filtering.

```
def filter_to_clinvar_pathogenic(
    t: Union[hl.MatrixTable, hl.Table],
    clnrevstat_expr: hl.ArrayExpression, 
    clnsig_expr: hl.ArrayExpression, 
    clnsigconf_expr: hl.ArrayExpression,
    remove_no_assertion: bool = True,
    remove_conflicting: bool = True,
) -> Union[hl.MatrixTable, hl.Table]:
    """
    docstring...
    """
    logger.info(f"Found {t.count_rows() if isinstance(t, hl.MatrixTable) else t.count()} variants before filtering")
    path_expr = t.info.CLNSIG.map(lambda x: x.lower()).map(lambda x: x.contains("pathogenic")).any(lambda x: x)
    
    if remove_no_assertion:
        path_expr = path_expr & (hl.set(t.info.CLNREVSTAT).intersection(no_star_assertions).length() == 0) 
        no_star_assertions = hl.literal(
            {
                "no_assertion_provided",
                "no_assertion_criteria_provided",
                "no_interpretation_for_the_single_variant",
            }
        )
    
    if remove_conflicting:
        path_expr = path_expr & hl.is_missing(t.info.CLNSIGCONF)

    if isinstance(t, hl.MatrixTable):
        t = t.filter_rows(path_expr)
    else:
        t = t.filter(path_expr)

    logger.info(
        f"Found {t.count_rows() if isinstance(t, hl.MatrixTable) else t.count()} variants after filtering to clinvar pathogenic variants"
    )

    return t
```

Use in this function would then be:
```
clinvar_path_ht = filter_to_clinvar_pathogenic(
    clinvar_ht,
    clinvar_ht.info.CLNREVSTAT, 
    clinvar_ht.info.CLNSIG, 
    clinvar_ht.info.CLNSIGCONF,
)
```

Reply:
I like this idea best and I know the above function needs to be cleaned up a bit but I think adding defaults for 
```
clnrevstat_expr: hl.ArrayExpression, 
clnsig_expr: hl.ArrayExpression, 
clnsigconf_expr: hl.ArrayExpression,``` 

in the final version would make sense since the I think we can expect us/users to annotate a HT/MT with our clinvar resource. Alternatively, we can keep them optional and adjust the code to not rely on them.

Reply:
@mike-w-wilson I fully agree it would be great to have defaults. Probably a question I should know the answer to, but can we add defaults for hail expressions? I have always switched to strings of column names to add defaults which works, but I do like keeping it as expressions if possible.

Reply:
@jkgoodrich,  I've been thinking about how to best implement this this afternoon and haven't settled on anything I like. I guess I should have a better understanding of how hail handles these things esp. lazy evaluation vs repeated evaluations and what triggers what. Perhaps this is a better question for zulip or discuss.hail.is

Reply:
great idea Mike, @williamphu can you post on discuss.hail.is. If you give a link to the discussion (which is public) they can see what we are trying to do.

Reply:
Yep, I support extracting the code out to a more general function. The hail discussion is here: https://discuss.hail.is/t/pass-default-for-hail-expressions-in-function/1874.

To double check, would `filter_to_clinvar_pathogenic` only be applied to clinvar resources (expect users to pass in a clinvar), or would any table with corresponding columns be filterable?

Reply:
I think it is likely to only be applied to clinvar resources, but the above suggestion would make it more general and could then be applied to any table with the needed columns.

Reply:
Based on the comments from the hail discuss, we'll use the string name of the column for the defaults.

Reply:
New code pushed to reflect this restructuring.